### PR TITLE
Update Axom to latest, enable monitoring in sundials

### LINKED ISSUE #355: Add sundials monitoring build flag
From @jandrej in the Teams chat: 

Can I request a small change in the Sundials TPL? We should enable (the recently changed) build system flag SUNDIALS_BUILD_WITH_MONITORING to allow printing in linear iterations in KINSOL. If someone with the machinery for it could rebuild tpls that would be greatly appreciated

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Most of this PR is changes to the input files to use vectors instead of individual components.  This feature is brand-new in Inlet, so any thoughts on style/usability might be good in case we want to tweak/add functionality to the Lua usertype that Inlet creates for its new vector type.

Resolves #355

### CODE REVIEW

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua [hunk 1]
```diff
@@ -54,15 +54,15 @@ nonlinear_solid = {
 
     -- initial conditions
     initial_displacement = {
-        vec_coef = function (x, y, z)
-            return 0, 0, 0
-        end  
+        vec_coef = function (v)
+            return Vector.new(0, 0, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua]: Will this work if it's just `Vector.new(0,0)`? The newmark problems are actually 2D.

**Comment by joshessman-llnl** [comment]
We can remove the copy of the sundials package if/when https://github.com/spack/spack/pull/21429 is merged.

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua [hunk 2]
```diff
@@ -54,15 +54,15 @@ nonlinear_solid = {
...
+        end
     },
 
     initial_velocity = {
-        vec_coef = function (x, y, z)
-            return 0, 1, 0
-        end 
+        vec_coef = function (v)
+            return Vector.new(0, 1, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua]: This one is actually 2D also.

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending.lua [hunk 3]
```diff
@@ -76,19 +76,19 @@ nonlinear_solid = {
 
     -- initial conditions
     initial_displacement = {
-        vec_coef = function (x, y, z)
-            return 0, 0, 0
-        end  
+        vec_coef = function (v)
+            return Vector.new(0, 0, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending.lua]: Also a 2D Vector.

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending.lua [hunk 4]
```diff
@@ -76,19 +76,19 @@ nonlinear_solid = {
...
     -- boundary condition parameters
     boundary_conds = {
         ['displacement'] = {
             -- boundary attribute 1 (index 0) is fixed (Dirichlet) in the x direction
             attrs = {2},
-            vec_coef = function (x, y, z)
-                return 0, 0, 0
-            end 
+            vec_coef = function (v)
+                return Vector.new(0, 0, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending.lua]: Also a 2D Vector.

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending_first.lua [hunk 5]
```diff
@@ -76,19 +76,19 @@ nonlinear_solid = {
 
     -- initial conditions
     initial_displacement = {
-        vec_coef = function (x, y, z)
-            return 0, 0, 0
-        end  
+        vec_coef = function (v)
+            return Vector.new(0, 0, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending_first.lua]: Also a 2D Vector.

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending_first.lua [hunk 6]
```diff
@@ -76,19 +76,19 @@ nonlinear_solid = {
...
     -- boundary condition parameters
     boundary_conds = {
         ['displacement'] = {
             -- boundary attribute 1 (index 0) is fixed (Dirichlet) in the x direction
             attrs = {2},
-            vec_coef = function (x, y, z)
-                return 0, 0, 0
-            end 
+            vec_coef = function (v)
+                return Vector.new(0, 0, 0)
```
Comment (jonwong12) [inline data/input_files/tests/nonlinear_solid/dyn_newmark_solve_bending_first.lua]: Also a 2D Vector.

File: data/input_files/default.lua [hunk 7]
```diff
@@ -54,29 +54,29 @@ nonlinear_solid = {
...
         },
         ['traction'] = {
             attrs = {2},
-            vec_coef = function (x, y, z)
-                return 0, 1.0e-3, 0
+            vec_coef = function (v)
+                return Vector.new(0, 1.0e-3, 0)
             end
             -- FIXME: Move time-scaling logic to Lua once arbitrary function signatures are allowed
             -- vec_coef = function (x, y, z, t)
```
Comment (white238) [inline data/input_files/default.lua]: Can you swap this commented function to the new class?

File: scripts/spack/packages/sundials/package.py [hunk 8]
```diff
@@ -0,0 +1,625 @@
...
+            '-DBUILD_SHARED_LIBS=%s' % on_off('+shared'),
+            '-DBUILD_STATIC_LIBS=%s' % on_off('+static')
+        ])
+
+        # generic (std-c) math libraries
+        args.extend([
+            '-DUSE_GENERIC_MATH=%s' % on_off('+generic-math')
+        ])
+
+        # SERAC EDIT BEGIN
```
Comment (white238) [inline scripts/spack/packages/sundials/package.py:278]: 👍

**Review by white238 — APPROVED** [review]

File: data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua (hunk 1 above)
Comment (joshessman-llnl) [reply data/input_files/tests/nonlinear_solid/dyn_newmark_solve.lua]: Yes - for reference, I've got the full set of permissible operations documented here: https://axom.readthedocs.io/en/develop/axom/inlet/docs/sphinx/advanced_types.html#adding-functions-to-a-schema

File: data/input_files/tests/nonlinear_solid/dyn_linesearch_solve.lua [hunk 9]
```diff
@@ -59,29 +59,33 @@ nonlinear_solid = {
...
-        vec_coef = function (x, y, z)
+        vec_coef = function (v)
+            x = v.x
             s = 0.1 / 64
             first = -s * x * x
             last = s * x * x * (8.0 - x)
-            -- FIXME: How can we detect the dimension?
-            return first, 0, last
-        end 
+            if v.dim == 2 then
```
Comment (jamiebramwell) [inline data/input_files/tests/nonlinear_solid/dyn_linesearch_solve.lua:73]: I really like this!

File: src/serac/infrastructure/input.cpp [hunk 10]
```diff
@@ -102,9 +102,9 @@ void CoefficientInputOptions::defineInputFileSchema(axom::inlet::Table& table)
 {
   // Vectors are expanded to three arguments in Lua (x, y, z)
   // and should be returned as a 3-tuple
-  table.addFunction("vec_coef", axom::inlet::FunctionTag::Vec3D, {axom::inlet::FunctionTag::Vec3D},
+  table.addFunction("vec_coef", axom::inlet::FunctionTag::Vector, {axom::inlet::FunctionTag::Vector},
```
Comment (jamiebramwell) [inline src/serac/infrastructure/input.cpp:105]: 👍

**Review by jamiebramwell — APPROVED** [review]
I really like the new Inlet features!

**Comment by joshessman-llnl** [comment]
LGTM
